Assignment2-Ancestors/Q1.py

```diff
@@ -0,0 +1,44 @@
+from Tree import BinaryTree
+
+def find_ancestors(tree, key):
+    """
+    Find ancestors of the first node with the given value while preorder traversal and save the Nodes in a list that return.
+    :param tree: Tree
+    :param key: any
+    :return: list(Nodes)
+    """
+    node = tree.find(key)
+    if node:

```

Comment:
Returns `None` if node is not found? Maybe should be mentioned in the docstring above? Also possibly:

```
if not node:
  return None
parent = node.parent
...
```

So that the error handling case would be explicit.